app/jobs/scheduled_publishing_job.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+class ScheduledPublishingJob < ApplicationJob
+  # retry at 3s, 18s, 83s, 258s, 627s
+  retry_on(StandardError,
+           wait: :exponentially_longer,
+           attempts: 5)
+
+  discard_and_log(ActiveRecord::RecordNotFound)
+
+  def perform(id)
+    Edition.find_and_lock_current(id: id) do |edition|
+      next if no_longer_schedulable?(edition)
+
+      user = edition.status.created_by
+      reviewed = edition.status.details.reviewed
+
+      PublishService.new(edition.document)
+                    .publish(user: user, with_review: reviewed)
+    end
+  end
+
+private
+
+  def no_longer_schedulable?(edition)
+    unless edition.scheduled?
+      Rails.logger.warn("Cannot publish an edition that is not scheduled")
```

Comment:
Small thing that I missed last time, it'd probably be useful if this and the other log included the edition id for them to be something we can identify from a log.

Reply:
Sure - I've amended with the ID :-).

Reply:
Awesome 👍